jpegutils.c

```diff
@@ -452,6 +444,13 @@ int decode_jpeg_raw (unsigned char *jpeg_data, int len,
     int i, xsl, xsc, xs, hdown;
     unsigned int x, y = 0, vsf[3], xd;
 
+    unsigned char buf0[16][MAX_LUMA_WIDTH];
+    unsigned char buf1[8][MAX_CHROMA_WIDTH];
+    unsigned char buf2[8][MAX_CHROMA_WIDTH];
+    unsigned char chr1[8][MAX_CHROMA_WIDTH];
+    unsigned char chr2[8][MAX_CHROMA_WIDTH];
```

Comment:
If I did my maths right, we're putting 64K of data onto the stack. That seems like quite a large chunk. I think I've generally worked to a rule of thumb than anything over a few K usually shouldn't go on the stack.

The alternative I presume would be hanging malloced blocks off of dinfo.

Thoughts?